samcli/local/docker/container.py

```diff
@@ -235,8 +244,12 @@ def wait_for_result(self, name, event, stdout, stderr):
         # NOTE(sriram-mv): Let logging happen in its own thread, so that a http request can be sent.
         # NOTE(sriram-mv): All logging is re-directed to stderr, so that only the lambda function return
         # will be written to stdout.
-        logs_thread = threading.Thread(target=self.wait_for_logs, args=(stderr, stderr), daemon=True)
-        logs_thread.start()
+
+        # the log thread will not be closed until the container itself got deleted,
+        # so as long as the container is still there, no need to start a new log thread
+        if not self._logs_thread or not self._logs_thread.is_alive():
+            self._logs_thread = threading.Thread(target=self.wait_for_logs, args=(stderr, stderr), daemon=True)
+            self._logs_thread.start()
```

Comment:
Are we shutting down this thread in any where else? My worry is if these will keep spin-up, we might end-up with a memory leak.

Reply:
This thread will keep running as long as the container is running, and it will finish once the container got deleted.